File: 백준/Bronze/1145. 적어도 대부분의 배수/적어도 대부분의 배수.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] val = br.readLine().split(" ");
        ArrayList<Integer> li = new ArrayList<>(val.length);

        for (int i = 0; i < val.length; i++) li.add(Integer.parseInt(val[i]));

        Collections.sort(li);
        Integer num = li.get(0);
        while (true) {
            if (find(num, li)) {
                System.out.println(num);
                break;
            }
            num ++;
        }

    }
    public static Boolean find(Integer num, ArrayList<Integer> li) {
        for (int i = 0; i < 3; i++) {
            if (num % li.get(i) != 0) continue;
            for (int j = i+1; j < 4; j++) {
                if (num % li.get(j) != 0) continue;
                for (int k = j+1; k < 5; k++) {
                    if (num % li.get(k) == 0) return true;
                }
            }
        }
        return false;
    }
}

File: 백준/Bronze/1145. 적어도 대부분의 배수/적어도 대부분의 배수.py
import sys
input = sys.stdin.readline

def find(num):
    for i in range(3):
        if num%li[i]: continue
        for j in range(i+1,4):
            if num%li[j]: continue
            for k in range(j+1,5):
                if num%li[k]: continue
                else: return True
    return False

li = sorted(list(map(int,input().split())))
result = li[0]

while True:
    if find(result): 
        print(result)
        break
    else: result += 1